variants/STM32U5xx/U575C(G-I)(T-U)_U585CI(T-U)/variant_HERON.h

```diff
@@ -0,0 +1,246 @@
+ * License. You may obtain a copy of the License at:
+ *                        opensource.org/licenses/BSD-3-Clause
+ *
+ *******************************************************************************
+ */
+#pragma once
+
+/*----------------------------------------------------------------------------
+ *        STM32 pins number
+ *----------------------------------------------------------------------------
+ *        Arduino digital pin numbers on the right (indexes into the digitalPin[] array)
+ *        and the STM32 pins they correspond to on the left.  The only apparent
+ *        function I can see is to reference a pin in a sketch in something
+ *        like digitalWrite() and have it index into digitalPin[] to find the
+ *        actual pin.  On the other hand, PIN_A* are numbers offset by PNUM_ANALOG_BASE,
+ *        which serves to say that this IS an analog pin and which tells analogWrite
+ *        to index into the digitalPin[] array to find the actual pin.
+ *----------------------------------------------------------------------------*/
+#define PA0                       PIN_A0  // A0/D0
+#define PA1                       PIN_A1  // A1/D1
+#define PA2                       PIN_A2  // A2/D2
+#define PA3                       PIN_A3  // A3/D3
+#define PA4                       PIN_A8  // A8/BATTERY_VOLTAGE (STAT)
+#define PA5                       25      // USART3_VCP_RX
+#define PA6                       PIN_A6  // A6/MI
+#define PA7                       PIN_A5  // A5
+#define PA8                       7       // LED_BUILTIN
+#define PA9                       19      // TX
+#define PA10                      18      // RX
+#define PA11                      32      // USB_D_N
+#define PA12                      33      // USB_D_P
+#define PA13                      23      // SWDIO
+#define PA14                      24      // SWCLK
+#define PA15                      29      // CHARGE_DETECT
+#define PB0                       PIN_A7  // A7/D11
+#define PB1                       PIN_A4  // A4/D4
+#define PB2                       31      // USB_DETECT
+#define PB3                       15      // CK
+#define PB4                       13      // D13
+#define PB5                       16      // MO
+#define PB6                       21      // SCL
+#define PB7                       22      // SDA
+#define PB8                       5       // D5
+#define PB9                       6       // D6
+#define PB10                      26      // USART3_VCP_TX
+#define PB13                      10      // D10
+#define PB14                      9       // D9
+#define PB15                      12      // D12
+#define PC13                      8       // USER_BTN
+#define PC14                      34      // OSC32_IN (LSE)
+#define PC15                      35      // OSC32_OUT (LSE)
+#define PH0                       27      // ENABLE_3V3
+#define PH1                       28      // DISCHARGE_3V3
+#define PH3                       20      // B
```

Comment:
Hard to check if all pin have an unique number as not ordered. Anyway I guess you have tested.
You can use this example to check if definition is correct:
https://github.com/stm32duino/STM32Examples/tree/main/examples/NonReg/CheckVariant